tests/AutoReview/ProjectCodeTest.php

```diff
@@ -264,6 +265,21 @@ public function testThatTestClassesAreInternal($className)
         );
     }
 
+    /**
+     * @dataProvider provideTestClassNamesAndDataProviderMethodNames
+     *
+     * @param string $testClassName
+     * @param string $dataProviderMethodName
+     */
+    public function testThatDataProvidersAreNotPrefixedWithTest($testClassName, $dataProviderMethodName)
+    {
+        $this->assertNotRegExp('/^test/', $dataProviderMethodName, sprintf(
```

Comment:
please match the `/^provide\S+Cases/$`

Reply:
Not sure I understand, would you like me to assert something like

```php
$this->assertRegExp('/^provide[A-Z]\S+Cases$/', $dataProviderMethodName);
```

instead of what I have here?

Reply:
yep
so,instead of ensuring that test method is not starting with sth,
ensure it does match whole format